tests/integ/test_inheritance.py

```diff
@@ -0,0 +1,209 @@
+import uuid
+from datetime import datetime, timezone
+from string import ascii_letters
+
+import pytest
+import random
+
+from bloop import BaseModel, Column, DateTime, GlobalSecondaryIndex, Integer, UUID
+from bloop.exceptions import InvalidIndex, InvalidModel
+from tests.integ.models import ExternalUser, MixinUser, Role
+
+
+def test_inheritance_simple(engine):
+    class NewBase(BaseModel):
+        class Meta:
+            abstract = True
+        uuid = Column(UUID)
+
+    class SimpleModel(NewBase):
+        id = Column(Integer, hash_key=True)
+        created_at = Column(DateTime)
+
+    model = SimpleModel()
+    assert len(model.Meta.columns) == 3
+    assert len(model.Meta.keys) == 1
+    assert list(model.Meta.keys)[0].name == 'id'
+
+
+def test_inheritance_base_hashkey(engine):
+    class NewBase(BaseModel):
+        class Meta:
+            abstract = True
+        uuid = Column(UUID, hash_key=True)
+
+    class SimpleModel(NewBase):
+        id = Column(Integer)
+        created_at = Column(DateTime)
+
+    model = SimpleModel()
+    assert len(model.Meta.columns) == 3
+    assert len(model.Meta.keys) == 1
+    assert list(model.Meta.keys)[0].name == 'uuid'
+
+
+def test_inheritance_mixins(engine):
+    model = MixinUser()
+    assert len(model.Meta.columns) == 8
+    assert len(model.Meta.keys) == 2
+    assert model.Meta.hash_key.name == 'id'
+    assert model.Meta.range_key.name == 'created'
+
+
+def _get_user(cls, **extra):
+    now = datetime.now(timezone.utc)
+    first_name = "".join([random.choice(ascii_letters) for _ in range(8)])
+    last_name = "".join([random.choice(ascii_letters) for _ in range(12)])
+    email = f"{first_name}.{last_name}@example.com"
+
+    return cls(
+        id=uuid.uuid4(), created=now, updated=now, active=True,
+        first_name=first_name, last_name=last_name, email=email,
+        **extra
+    )
+
+
+def gen_external_user():
+    extra = {'company': 'Acme', 'roles': {Role.user, Role.admin}}
+    return _get_user(ExternalUser, **extra)
+
+
+def gen_mixin_user():
+    extra = {'roles': {Role.user}}
+    return _get_user(MixinUser, **extra)
+
+
+@pytest.mark.parametrize("cls, factory", [
+    (MixinUser, gen_mixin_user),
+    (ExternalUser, gen_external_user)
+])
+def test_inheritance_load(engine, cls, factory):
+    engine.bind(BaseModel)
+
+    model = factory()
```

Comment:
nit: so far we've used `model` to mean a subclass of `BaseModel`, while this is an instance.  I've used `obj` and then later `same_obj` when I want to query that thing again.